File: python/006.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import math

def powsum(n):
    return sum([int(math.pow(x, 2)) for x in range(1, n)])

def sumpow(n):
    return int(math.pow(sum([ x for x in range(1, n)]), 2))

def p6(n):
    return sumpow(n) - powsum(n)

def main():
    print(p6(100))

if __name__ == '__main__':
    main()

File: elixir/006.exs
#!/usr/bin/env elixir

# https://projecteuler.net/problem=6

defmodule P6 do
	def powsum(n) do
		1..n
		|> Enum.map(&( &1 * &1 ))
		|> Enum.sum
	end
	def sumpow(n) do
		s = Enum.sum(1..n)
		s * s
	end
end

P6.sumpow(100) - P6.powsum(100)
|> IO.inspect
